src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/constrainedtransfer/TransferIndexGenerator.java

```diff
@@ -227,17 +231,69 @@ private List<TPoint> findTPointForRoute(
 
   private List<TPoint> findTPoints(TripTransferPoint point) {
     var trip = point.getTrip();
-    // All trips have at least one pattern, no need to chech for null here
+
+    // All trips have at least one pattern, no need to check for null here
     var patterns = patternsByTrip.get(trip);
-    int stopPosInPattern = point.getStopPositionInPattern();
-    return patterns
+    var groupedPatterns = patterns
       .stream()
-      .map(pattern -> {
-        int stopIndex = pattern.stopIndex(stopPosInPattern);
-        var sourcePoint = createTransferPointForPattern(trip, stopIndex);
-        return new TPoint(pattern, sourcePoint, trip, stopPosInPattern);
-      })
-      .toList();
+      .collect(Collectors.groupingBy(pattern -> pattern.getPattern().isCreatedByRealtimeUpdater()));
+
+    // Process first the pattern for which stopPosInPattern was calculated for
+    List<RoutingTripPattern> scheduledPatterns = groupedPatterns.get(Boolean.FALSE);
+    if (scheduledPatterns == null || scheduledPatterns.size() != 1) {
+      LOG.warn(
+        "Trip {} does not have exactly one scheduled trip pattern, found: {}. " +
+        "Skipping transfer generation.",
+        trip.getId(),
```

Comment:
```suggestion
        trip,
```
Trip implements LogInfo and inherit a "safe" `toString()` ment for logging.

Reply:
Done